File: LeetCode/901-1000/941. Valid Mountain Array/src/Solution.java
class Solution {
    public boolean validMountainArray(int[] arr) {
        if (arr.length < 3 || !(arr[0] < arr[1])) {
            return false;
        }
        boolean bool = false;
        for (int i = 0; i < arr.length - 1; i++) {
            if (!bool) {
                if (arr[i] < arr[i + 1]) {
                    //
                } else if (arr[i] == arr[i + 1]) {
                    return false;
                } else {
                    if (arr[i - 1] < arr[i] && arr[i] > arr[i + 1]){
                        bool = true;
                    }
                }
            } else {
                if (arr[i] > arr[i + 1]) {
                    //
                } else {
                    return false;
                }
            }
        }
        return bool;
    }
}

File: LeetCode/901-1000/941. Valid Mountain Array/Solution.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        int n = arr.size();
        int pointer = 0;
        while (pointer < n - 1) {
            if (arr[pointer] < arr[pointer + 1]) {
                pointer++;
            } else {
                break;
            }
        }
        if (pointer == 0 || pointer == n - 1) {
            return false;
        }
        while (pointer < n - 1) {
            if (arr[pointer] > arr[pointer + 1]) {
                pointer++;
            } else {
                return false;
            }
        }
        return true;
    }
};